Fix tests

myc2h6o opened the pull request:
- Resource config definition update
  - Move `resource "azurerm_key_vault_access_policy" "disk-encryption" { ...` block to `dependencies` config, to avoid the duplicate definition of this resource. Earlier this resource is only needed for key rotate, but now the update of the new flag `auto_key_rotation_enabled` also requires this, and since in the [document](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/website/docs/r/disk_encryption_set.html.markdown#example-usage), it is already present, I'm moving it to the common resource dependencies, to make it simple to write the update and keyRotate test
  - Here is failed test output before this change
```
------ Stdout: -------
=== RUN   TestAccDiskEncryptionSet_update
=== PAUSE TestAccDiskEncryptionSet_update
=== CONT  TestAccDiskEncryptionSet_update
testcase.go:108: Step 3/6 error: Error running apply: exit status 1
Error: updating Disk Encryption Set "acctestDES-211019081335747893" (Resource Group "acctestRG-211019081335747893"): compute.DiskEncryptionSetsClient#Update: Failure sending request: StatusCode=400 -- Original Error: Code="KeyVaultAccessForbidden" Message="Unable to access key vault 'https://acctestkv-rzdvo.vault.azure.net/keys/examplekey'"
with azurerm_disk_encryption_set.test,
on terraform_plugin_test.tf line 69, in resource "azurerm_disk_encryption_set" "test":
69: resource "azurerm_disk_encryption_set" "test" {
--- FAIL: TestAccDiskEncryptionSet_update (385.25s)
FAIL
```
- TestAccDataSourceDiskEncryptionSet_basic
  - Remove the duplicate basic config
  - Add check for default value of computed auto_key_rotation_enabled
- TestAccDiskEncryptionSet_update
  - Remove duplicated complete config
- TestAccDiskEncryptionSet_keyRotate
  - Remove grantAccessToKeyVault config after the move of `azurerm_key_vault_access_policy` resource, as it is now a common resource in `dependencies`, which will be present across all config.
- Test results:
```
$ make acctests SERVICE='compute' TESTARGS='-run=DiskEncryptionSet_' TESTTIMEOUT='60m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/compute -run=DiskEncryptionSet_ -timeout 60m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccDataSourceDiskEncryptionSet_basic
=== PAUSE TestAccDataSourceDiskEncryptionSet_basic
=== RUN   TestAccDataSourceDiskEncryptionSet_update
=== PAUSE TestAccDataSourceDiskEncryptionSet_update
=== RUN   TestAccDiskEncryptionSet_basic
=== PAUSE TestAccDiskEncryptionSet_basic
=== RUN   TestAccDiskEncryptionSet_requiresImport
=== PAUSE TestAccDiskEncryptionSet_requiresImport
=== RUN   TestAccDiskEncryptionSet_complete
=== PAUSE TestAccDiskEncryptionSet_complete
=== RUN   TestAccDiskEncryptionSet_update
=== PAUSE TestAccDiskEncryptionSet_update
=== RUN   TestAccDiskEncryptionSet_keyRotate
=== PAUSE TestAccDiskEncryptionSet_keyRotate
=== CONT  TestAccDataSourceDiskEncryptionSet_basic
=== CONT  TestAccDiskEncryptionSet_complete
=== CONT  TestAccDiskEncryptionSet_keyRotate
=== CONT  TestAccDataSourceDiskEncryptionSet_update
=== CONT  TestAccDiskEncryptionSet_update
=== CONT  TestAccDiskEncryptionSet_basic
=== CONT  TestAccDiskEncryptionSet_requiresImport
--- PASS: TestAccDiskEncryptionSet_complete (530.87s)
--- PASS: TestAccDiskEncryptionSet_basic (532.79s)
--- PASS: TestAccDataSourceDiskEncryptionSet_basic (545.35s)
--- PASS: TestAccDiskEncryptionSet_requiresImport (620.24s)
--- PASS: TestAccDataSourceDiskEncryptionSet_update (712.04s)
--- PASS: TestAccDiskEncryptionSet_update (726.25s)
--- PASS: TestAccDiskEncryptionSet_keyRotate (755.83s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/compute       756.524s
```

myc2h6o committed:
Fix tests

Tbohunek commented:
I'm really sorry you have to keep working through my repo.

However, are you sure reducing those test cases preserves original functionality? I don't see those tests as duplicate, rather testing different scenarios.
Furthermore, you're even removing code/tests that were already there in Terraform before my PR and that exist also in other resources such as Key Vault.

myc2h6o commented:
No worries.
I'm not reducing the test cases, just moving the common key vault access policies assignment resource to the dependencies blocks.
You can see I removed `func (r DiskEncryptionSetResource) grantAccessToKeyVault(data acceptance.TestData) string`, which is originally meant to assign the access policies of the key vault to the disk encryption set. I moved it to `func (DiskEncryptionSetResource) dependencies(data acceptance.TestData) string`, which is the common part for all other config function.
The reason for this is that: updating the flag `enable_auto_key_rotation` requires a key vault access policy as well, so the update scenario will also need it. Instead of adding the duplicate assignment to the update test, I moved it to the common part.

myc2h6o commented on internal/services/compute/disk_encryption_set_resource_test.go:
```diff
@@ -101,16 +94,6 @@ func TestAccDiskEncryptionSet_keyRotate(t *testing.T) {
 			),
 		},
-		data.ImportStep(),
-		// we have to first grant the permission for DiskEncryptionSet to access the KeyVault
-		{
-			Config: r.grantAccessToKeyVault(data),
-			Check: acceptance.ComposeTestCheckFunc(
-				check.That(data.ResourceName).ExistsInAzure(r),
-			),
-		},
-		data.ImportStep(),
```
This part is used to add the below resource, and it is moved to the `dependencies` function, and will be present when `basic` step is run. The functionality will remain same
```
resource "azurerm_key_vault_access_policy" "disk-encryption" {
  key_vault_id = azurerm_key_vault.test.id
  key_permissions = [
    "Get",
    "WrapKey",
    "UnwrapKey",
  ]
  tenant_id = azurerm_disk_encryption_set.test.identity.0.tenant_id
  object_id = azurerm_disk_encryption_set.test.identity.0.principal_id
}
```
